src/main.cpp

```diff
@@ -48,7 +48,7 @@ int main(int argc, const char *argv[]) {
 
   //write the results into a hdf5 file
   if (c.rank() == 0) {
-    std::string output_file = par["outputfile"];
+       std::string output_file = par["outputfile"].template as<std::string>();
```

Comment:
In this case, `par["outputfile"]` should unambiguously be converted to `std::string` automatically, so `as<std::string>()` is unnecessary (and i believe `template` is not needed either).

Reply:
I understand. I will modify following your recommendations and re-submit. Thanks